### PR TITLE
Add "Your Last Location" to Android Home

### PR DESCRIPTION
**Pull request by gcalero**
Changes in Android init behaviour
+ Not connecting to any domain at startup
+ New option "Your Last Domain" in the Home screen
+ We avoid audio issues at startup

Test plan:
- Uninstall interface from your phone before downloading this PR.
- Open Interface.  You should see "Your Last Location" at the top of the list of domains.  Go to any domain and have someone else go there (on desktop is fine).  Verify that they hear you when you speak.
- Close Interface (the other person remains in that domain).  Open Interface.  The other person shouldn't see you appear, or be able to hear you.
- Press "Your Last Location".  You should appear next to the other person, and they should be able to hear you.
- Domain images should appear correctly, and there should be a new image for domains with the default image.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9918/HighFidelity-Beta-PR13197-7a6b7b8ba3207b6518515d37ccfca8c72afaa141.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9918/HighFidelity-Beta-PR13197-7a6b7b8ba3207b6518515d37ccfca8c72afaa141.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9918/HighFidelity-Beta-Interface-PR13197-7a6b7b8ba3207b6518515d37ccfca8c72afaa141.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9918/HighFidelity-Beta-PR13197-7a6b7b8ba3207b6518515d37ccfca8c72afaa141.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9918/HighFidelity-Beta-Interface-PR13197-7a6b7b8ba3207b6518515d37ccfca8c72afaa141.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9938/HighFidelity-Beta-PR13197-85cff19020d2b37e21c2c863176f747728d2c5fc.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9938/HighFidelity-Beta-PR13197-85cff19020d2b37e21c2c863176f747728d2c5fc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9938/HighFidelity-Beta-Interface-PR13197-85cff19020d2b37e21c2c863176f747728d2c5fc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9938/HighFidelity-Beta-PR13197-85cff19020d2b37e21c2c863176f747728d2c5fc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9938/HighFidelity-Beta-Interface-PR13197-85cff19020d2b37e21c2c863176f747728d2c5fc.exe

File: android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java [hunk 1]
```diff
@@ -51,9 +53,29 @@ public void loadDomains(String filterText) {
...
+                if (!mLastLocation.isEmpty() && mLastLocation.contains("://")) {
+                    int startIndex = mLastLocation.indexOf("://");
+                    int endIndex = mLastLocation.indexOf("/", startIndex + 3);
+                    String toSearch = mLastLocation.substring(0, endIndex + 1).toLowerCase();
+                    for (Domain d : domain) {
+                        if (d.url.toLowerCase().startsWith(toSearch)) {
+                            lastVisitedDomain.thumbnail = d.thumbnail;
+                        }
+                    }
+                }
```
Comment (SamGondelman) [inline android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java]: This whole block can be inside the `if (filterText.length() == 0)` check below.

File: android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java [hunk 2]
```diff
@@ -51,9 +53,29 @@ public void loadDomains(String filterText) {
...
+
                 if (filterText.length() == 0) {
-                    domain.add(0, new Domain(mContext.getString(R.string.your_last_location), "", DEFAULT_THUMBNAIL_PLACE));
+                    domain.add(0, lastVisitedDomain);
                 }
+
+                for (Domain d : domain) {
+                    if (d.thumbnail != null &&
+                            d.thumbnail.endsWith("assets/places/thumbnail-default-place-e5a3f33e773ab699495774990a562f9f7693dc48ef90d8be6985c645a0280f75.png")) {
```
Comment (SamGondelman) [inline android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java]: We can just do `d.thumbnail.startsWith("/")`, right?

Comment (SamGondelman) [reply android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java]: Also, can you leave a comment above this explaining that we're replacing the picture?

Comment (gcalero) [reply android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java]: L74: at this point url's are sanitized so it doesn't start with /assets/...

Comment (gcalero) [reply android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java]: Adding the comment

File: android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java (hunk 1 above)
Comment (gcalero) [reply android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java]: I think so. Trying.

File: android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java (hunk 2 above)
Comment (gcalero) [reply android/app/src/main/java/io/highfidelity/hifiinterface/view/DomainAdapter.java]: QA, can you test this please?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9943/HighFidelity-Beta-PR13197-054e6041af6ff19dafc2b3e46052d2bbb12c49b5.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9943/HighFidelity-Beta-PR13197-054e6041af6ff19dafc2b3e46052d2bbb12c49b5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9943/HighFidelity-Beta-Interface-PR13197-054e6041af6ff19dafc2b3e46052d2bbb12c49b5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9943/HighFidelity-Beta-PR13197-054e6041af6ff19dafc2b3e46052d2bbb12c49b5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9943/HighFidelity-Beta-Interface-PR13197-054e6041af6ff19dafc2b3e46052d2bbb12c49b5.exe

**Comment by qligmaloney** [comment]
Currently testing on Android and Desktop Interface

**Comment by qligmaloney** [comment]
PASS on Android and Desktop Interface @conklin94122 
My Last Location displays on Home Screen. User can hear each other speak while in the same domain, but not after application is closed on Android device. Android Smoke Test PASS

**Pull request merged**